hydrolib/core/basemodel.py

```diff
@@ -605,6 +650,24 @@ def resolve_casing(self, file_path: Path) -> Path:
             return self._file_casing_resolver.resolve(file_path)
         return file_path
 
+    def convert_path_style(self, file_path: Path) -> Path:
+        """Resolve the file path by converting it from its own file path style to the path style for the current operating system.
+
+        Args:
+            file_path (Path): The file path to convert to the OS path style.
+
+        Returns:
+            Path: The resolved file path.
+        """
+
+        if file_path.is_absolute():
```

Comment:
if `file_path.is_absolute()` we know for sure that the path can be interpreted correctly (this will happen for the rootmodel).
When we pass this check the file path is either:

- A relative path on the same OS
- A relative path on the opposite OS
- An absolute path on the opposite OS

But I see that in the `FilePathStyleConverter._convert` function that the first checks if the current OS is the source OS. Here it will just return the original file path.

So in the end I removed the `file_path.is_absolute()` check since it is not necessary.

Reply:
Edit: I did NOT remove the check. I made a thinking mistake. 
When a relative path is passed to the initializer of the file model, the converting functionality will indeed be called, but the result should be the same (so functionality wise it is correct). We do not know whether we are calling from the root model or a referenced sub model, only the `file_path.is_absolute()` check suffices for that when the root model is created with an absolute. Since performance and functionality wise it is correct, I propose to leave it like this instead of creating a way to derive whether we are calling from the root model or not.

Do you agree?